updateengine/status.go

```diff
@@ -42,6 +42,32 @@ func NewStatus(body []interface{}) (s Status) {
 	return
 }
 
+func IsSignalStatus(body []interface{}) (isStatus bool) {
+	if len(body) < 5 {
+		return false
+	}
+
+	// Check that elements of body have correct type corresponding to type Status
+	typeIsCorrect := false
+	for i, elem := range body {
```

Comment:
This code is somewhat awkward. I would suggest instead to move the nil+length+types checking directly into a fallible constructor (e.g. `TryNewStatus` or similar), as this file is the only consumer.
It would look like `TryNewStatus(*dbus.Signal) (Status, error)` so that you can `continue` on error in the loop above.

Reply:
I see - makes sense to make use of `error`. Will work on this now. Thank you!